autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/MarathonServiceScaler.java

```diff
@@ -70,11 +71,10 @@ public void scaleUp(final String serviceReference, final int amount)
             int current = app.getTasksRunning() + app.getTasksStaged();
             int target = Math.min(maximumInstances, current + amount);
             if ( target > current ) {
-                app.setInstances(Math.min(maximumInstances, app.getTasksRunning() + app.getTasksStaged() + amount));
                 LOG.debug("Scaling service {} up by {} instances", serviceReference, amount);
-                marathon.updateApp(serviceReference, app, true);
+                appInstancePatcher.patchInstances(app.getId(), target);
             }
-        } catch (MarathonException e) {
+        } catch (Exception e) {
```

Comment:
Don't catch the top-level `Exception`.  What type of exception are you actually trying to map?